### ISSUE TITLE
parse failure with LL(1) language of two custom entries

### ISSUE THREAD

**Issue by andres-erbsen** [coq/coq#9517]
#### Description of the problem

I would expect the following to succeed, but it does not.

```coq
Declare Custom Entry expr.
Declare Custom Entry stmt.
Notation "x" := x (in custom stmt, x ident).
Notation "x" := x (in custom expr, x ident).

Notation "1" := 1 (in custom expr).

Notation "! x = y !" := (pair x y) (in custom stmt at level 0, x custom expr, y custom expr).
Notation "? x = y" := (pair x y) (in custom stmt at level 0, x custom expr, y custom expr).
Notation "x = y" := (pair x y) (in custom stmt at level 0, x custom expr, y custom expr).

Notation "stmt:( s )" := s (s custom stmt).
Check stmt:(! _ = _ !).
Check stmt:(? _ = _).
Check stmt:(_ = _).
Check stmt:(! 1 = 1 !).
Check stmt:(? 1 = 1).
(* > Check stmt:(? 1 = 1). *)
(* >                   ^ *)
(* Error: Syntax error: [constr:stmt] expected after '=' (in [constr:stmt]). *)
Check stmt:(1 = 1).
(* > Check stmt:(1 = 1). *)
(* >             ^ *)
(* Syntax error: [constr:stmt level 200] expected after 'stmt:(' (in [constr:operconstr]). *)
Check stmt:(_ = 1).
(* > Check stmt:(_ = 1). *)
(* >                 ^ *)
(* Error: Syntax error: [constr:stmt] expected after '=' (in [constr:stmt]). *)
```

I see no fundamental reason for this limitation. Is this a bug / can it be fixed? If not, could I have some guidance for rewriting the grammar to parse languages with productions like this?

Thanks,
Andres

#### Coq Version

8.9.0

@herbelin ?

**Comment by staffehn** [comment]
Hi, I gave this a go and this is how far I came with rewriting the grammar.
```coq
Declare Custom Entry expr.
Declare Custom Entry stmt.
Notation "x" := x (in custom stmt, x ident).
Notation "x" := x (in custom expr at level 0, x ident).

Notation "1" := 2 (in custom stmt).
Notation "1" := 1 (in custom expr at level 0).

Notation "! x = y !" := (pair x y) (in custom stmt at level 0, x custom expr, y custom expr).
Notation "? x = y" := (pair x y) (in custom stmt at level 0, x custom expr, y custom expr at level 0).
Notation "x = y" := (pair x y) (in custom stmt at level 0, x custom expr, y custom expr at level 0).

Notation "stmt:( s )" := s (s custom stmt).
Check stmt:(! _ = _ !).
Check stmt:(? _ = _).
Check stmt:(_ = _).
Check stmt:(! 1 = 1 !).
Check stmt:(? 1 = 1).
(*
stmt:( 1 = 1)
     : nat * nat
*)
Check stmt:(1 = 1).
(*
(2, 1)
     : nat * nat
*)
Check stmt:(_ = 1).
(*
(?y, 1)
     : ?A * nat
where
?A : [ |- Type]
?y : [ |- ?A]
*)
```

Very apparently, in the "x = y" rule, the "x custom expr" hint is completely ignored without warning/error,
since as you can see, the "1" get's parsed with the "stmt" rule, giving a value "2".

Instead of writing "y custom expr at level 0", an additional "right associativity" seems to do the same thing. Interestingly, any explicit level for the "y" seems to do, #9520 seems related.

**Comment by herbelin** [comment]
Same cause as #11331 and fixed by #11530.
